R/group_data.R

```diff
@@ -63,3 +63,21 @@ add_sum_of_views <- function(df, view_name_pattern, method, new_name = view_name
   df_plus_new_group <- bind_rows(df, new_group)
   return(df_plus_new_group)
 }
+
+#' @param df data.frame of daily unique pageviews and page paths with site ids for noms
+#'           expects pagepath, date, uniquePageviews and pageviews
+#' @return data frame with date, uniquePageviews and site_no
+group_by_site_id <- function(df) {
+  
+  date <- df$date[1]
+  
+  df <- df %>%
+    mutate(site_no = regmatches(df$pagePath, gregexpr("[[:digit:]]+", df$pagePath))) %>% #get all numeric values into new column
+    unnest(site_no) %>% #unlist list column
+    filter(nchar(as.character(site_no)) >= 8) %>% #keep values at 8 or more characters, likely erroneous if less
```

Comment:
this filters out date parts and other small numbers less than 8 digist that got extracted from the pagePath in the first regmatches

Reply:
looking at this, i probably need to change this so it filters out any number that is not 8 digits or 15 digits, correct? e.g.  9-14 digit site_no are not a thing?

Reply:
Right